src/Databases/DatabaseReplicatedWorker.cpp

```diff
@@ -308,4 +313,18 @@ bool DatabaseReplicatedDDLWorker::canRemoveQueueEntry(const String & entry_name,
     return entry_number + logs_to_keep < max_log_ptr;
 }
 
+void DatabaseReplicatedDDLWorker::updateLogPointer(const String & processed_entry_name)
+{
+    updateMaxDDLEntryID(processed_entry_name);
+    assert(max_id.load() == parse<UInt32>(getAndSetZooKeeper()->get(fs::path(database->replica_path) / "log_ptr")));
+}
```

Comment:
This function I assume initializes the `log_ptr` because of the hard assumption in assert right?
`processTask` should update the `max_id` afterwards so why not rename the function to `initializeLogPointer` and make it private.

Reply:
Yes, `initializeLogPointer` is better name, but it cannot be private, because it's called from `DatabaseReplicated::recoverLostReplica`. However, we can update log pointer in `DatabaseReplicatedDDLWorker::initializeReplication` after recovering replica.